### ISSUE TITLE
Inconsistencies between reports

### ISSUE THREAD

**Issue by cbandy** [sebastianbergmann/php-code-coverage#154]
There are a number of inconsistencies in the statistics given by different reports. The reports generated for this repository are a good start. The details below are for reports generated by 1.2.9 against 1.2.9.
- [ ] Clover: `build/logs/clover.xml`

`PHP_CodeCoverage` is the only class inside `PHP/CodeCoverage.php` so the numbers should match, right? However, `coveredmethods` and `coveredelements` differ by 5.

``` xml
<file name="PHP/CodeCoverage.php">
  <metrics loc="801" ncloc="575" classes="1" methods="23" coveredmethods="16" conditionals="0" coveredconditionals="0" statements="352" coveredstatements="295" elements="375" coveredelements="311"/>
</file>
<class name="PHP_CodeCoverage" namespace="global" fullPackage="CodeCoverage" category="PHP" package="CodeCoverage">
  <metrics methods="23" coveredmethods="21" conditionals="0" coveredconditionals="0" statements="352" coveredstatements="295" elements="375" coveredelements="316"/>
</class>
```
- [ ] HTML: `build/coverage/index.html`

This matches Clover.

```
                        Lines        Functions and Methods Classes and Traits
CodeCoverage.php  83.81%   295 / 352      69.57%   16 / 23     0.00%   0 / 1
```
- [ ] HTML: `build/coverage/CodeCoverage.php.html`

This matches the numbers for `CodeCoverage.php` in this report, but it does not match the numbers for this class in Clover.

```
                 Classes and Traits Functions and Methods       Lines
PHP_CodeCoverage       0.00%  0 / 1       69.57%  16 / 23  83.81%  295 / 352
```
- [ ] Text: `--coverage-text`

This matches Clover, but not HTML.

```
@CodeCoverage::PHP_CodeCoverage
  Methods:  91.30% (21/23)   Lines:  83.81% (295/352)
```

---

In summary,

| _Matches_ | Clover File | Clover Class | HTML File | HTML Class | Text Class |
| --- | --- | --- | --- | --- | --- |
| Clover File |  | :no_entry_sign: | :black_circle: | :black_circle: | :no_entry_sign: |
| Clover Class | :no_entry_sign: |  | :no_entry_sign: | :no_entry_sign: | :black_circle: |
| HTML File | :black_circle: | :no_entry_sign: |  | :black_circle: | :no_entry_sign: |
| HTML Class | :black_circle: | :no_entry_sign: | :black_circle: |  | :no_entry_sign: |
| Text Class | :no_entry_sign: | :black_circle: | :no_entry_sign: | :no_entry_sign: |  |
